The server has exited with an error code of: 2

**Environment information:**
 - AGI version: agi-1.0.0
 - Host OS: Ubuntu


**Bug description**
when I start agi I get 
The application encountered an error:
The server has exited with an error code of: 2 
Most functions in AGI are unavailable without a server. 
You can restart the server, exit AGI, or close the dialog to continue without a server.
Please check the logs for details.

**Reproduction steps**
start agi

**Stacktrace**
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xaa4e44]

goroutine 34 [running]:
github.com/google/gapid/core/app/crash.Crash(0x3b01d60, 0x73c8770)
	core/app/crash/crash.go:89 +0xb1
github.com/google/gapid/core/app/crash.handler()
	core/app/crash/crash.go:56 +0x52
panic(0x3b01d60, 0x73c8770)
	GOROOT/src/runtime/panic.go:967 +0x15d
github.com/google/gapid/core/os/android/adb.newDevice(0x4a258a0, 0xc00023c390, 0xc00007b549, 0x8, 0x2, 0x0, 0x0, 0x0)
	core/os/android/adb/device.go:255 +0x824
github.com/google/gapid/core/os/android/adb.scanDevices(0x4a258a0, 0xc00023c390, 0x0, 0x0)
	core/os/android/adb/device.go:291 +0x33d
github.com/google/gapid/core/os/android/adb.Devices(0x4a258a0, 0xc00023c390, 0x0, 0x0, 0x0, 0x0, 0x0)
	core/os/android/adb/device.go:106 +0x39
main.monitorAndroidDevices.func1(0xc000223798, 0x4a258a0, 0xc00023c390, 0xc00023c0c0)
	cmd/gapis/main.go:163 +0x76
main.monitorAndroidDevices(0x4a258a0, 0xc00023c390, 0xc00023c0c0, 0xc000223798)
	cmd/gapis/main.go:169 +0x4d
main.run.func1()
	cmd/gapis/main.go:122 +0x63
github.com/google/gapid/core/app/crash.Go.func1(0xc000237aa0)
	core/app/crash/crash.go:65 +0x49
created by github.com/google/gapid/core/app/crash.Go
	core/app/crash/crash.go:63 +0x3f

```

**Screenshots**

![image](https://user-images.githubusercontent.com/7438184/94540445-c5892d80-0246-11eb-88e9-4559a465473b.png)



**Additional debugging information**
[gapis.log](https://github.com/google/agi/files/5297849/gapis.log)
[gapic.log](https://github.com/google/agi/files/5297850/gapic.log)

--- a/core/os/android/adb/device.go
+++ b/core/os/android/adb/device.go
@@ -150,6 +150,11 @@ func newDevice(ctx context.Context, serial string, status bind.Status) (*binding
 		}
 	}
 
+	// Early bail out if we cannot get device information
+	if d.To.Configuration.Hardware == nil {
+		return nil, log.Errf(ctx, nil, "Cannot get device information")
+	}
+
 	// Collect the operating system version
 	if version, err := d.SystemProperty(ctx, "ro.build.version.release"); err == nil {
 		var major, minor, point int32
@@ -222,37 +227,38 @@ func newDevice(ctx context.Context, serial string, status bind.Status) (*binding
 		d.To.Configuration.AnglePackage = anglePackage
 	}
 
-	// If the VkRenderStagesProducer layer exist, we assume the render stages producer is
-	// implemented in the layer.
-	for _, l := range d.To.Configuration.GetDrivers().GetVulkan().GetLayers() {
-		if l.Name == "VkRenderStagesProducer" {
-			capability := d.To.Configuration.PerfettoCapability
-			if capability == nil {
-				capability = &device.PerfettoCapability{
-					GpuProfiling: &device.GPUProfiling{},
+	// Query infos related to the Vulkan driver
+	if d.To.Configuration.GetDrivers() != nil && d.To.Configuration.GetDrivers().GetVulkan() != nil {
+
+		// If the VkRenderStagesProducer layer exist, we assume the render stages producer is
+		// implemented in the layer.
+		for _, l := range d.To.Configuration.GetDrivers().GetVulkan().GetLayers() {
+			if l.Name == "VkRenderStagesProducer" {
+				capability := d.To.Configuration.PerfettoCapability
+				if capability == nil {
+					capability = &device.PerfettoCapability{
+						GpuProfiling: &device.GPUProfiling{},
+					}
+					d.To.Configuration.PerfettoCapability = capability
 				}
+				gpu := capability.GpuProfiling
+				gpu.HasRenderStageProducerLayer = true
+				gpu.HasRenderStage = true
+				break
 			}
-			gpu := capability.GpuProfiling
-			gpu.HasRenderStageProducerLayer = true
-			gpu.HasRenderStage = true
-			break
+		}
+
+		if version, err := d.DriverVersionCode(ctx); err == nil {
+			d.To.Configuration.Drivers.Vulkan.Version = strconv.Itoa(version)
 		}
 	}
 
-	if d.To.Configuration == nil ||
-		d.To.Configuration.Hardware == nil {
-		return nil, log.Errf(ctx, nil, "Cannot get device information")
-	}
 	if d.Instance().GetName() == "" {
 		d.Instance().Name = d.To.Configuration.Hardware.Name
 	}
 	if i := d.Instance(); i.ID == nil || allZero(i.ID.Data) {
-		// Generate an identifier for the device based on it's details.
+		// Generate an identifier for the device based on its details.
 		i.GenID()
-	}
-
-	if version, err := d.DriverVersionCode(ctx); err == nil {
-		d.To.Configuration.Drivers.Vulkan.Version = strconv.Itoa(version)
 	}
 
 	return d, nil

